test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -246,23 +235,23 @@ final case class Spec[-R, +E](caseValue: SpecCase[R, E, Spec[R, E]]) extends Spe
    */
   final def mapError[E1](f: E => E1)(implicit ev: CanFail[E], trace: Trace): Spec[R, E1] =
     transform[R, E1] {
-      case ExecCase(exec, spec)        => ExecCase(exec, spec)
-      case LabeledCase(label, spec)    => LabeledCase(label, spec)
-      case ScopedCase(scoped)          => ScopedCase[R, E1, Spec[R, E1]](scoped.mapError(_.map(f)))
-      case MultipleCase(specs)         => MultipleCase(specs)
-      case TestCase(test, annotations) => TestCase(test.mapError(_.map(f)), annotations)
+      case e: ExecCase[Spec[R, E1]]     => e
+      case l: LabeledCase[Spec[R, E1]]  => l
+      case ScopedCase(scoped)           => ScopedCase[R, E1, Spec[R, E1]](scoped.mapError(_.map(f)))
+      case m: MultipleCase[Spec[R, E1]] => m
+      case TestCase(test, annotations)  => TestCase(test.mapError(_.map(f)), annotations)
     }
 
   /**
    * Returns a new spec with remapped labels.
    */
   final def mapLabel(f: String => String)(implicit trace: Trace): Spec[R, E] =
     transform[R, E] {
-      case ExecCase(exec, spec)        => ExecCase(exec, spec)
-      case LabeledCase(label, spec)    => LabeledCase(f(label), spec)
-      case ScopedCase(scoped)          => ScopedCase[R, E, Spec[R, E]](scoped)
-      case MultipleCase(specs)         => MultipleCase(specs)
-      case TestCase(test, annotations) => TestCase(test, annotations)
+      case e: ExecCase[Spec[R, E]]         => e
+      case LabeledCase(label, spec)        => LabeledCase(f(label), spec)
+      case s: ScopedCase[R, E, Spec[R, E]] => s
+      case m: MultipleCase[Spec[R, E]]     => m
+      case t: TestCase[R, E]               => t
```

Comment:
Does something like this work? Same goes throughout
```suggestion
      case LabeledCase(label, spec) => LabeledCase(f(label), spec)
      case s                                         => s
```